transmart-core-db/grails-app/services/org/transmartproject/db/multidimquery/QueryService.groovy

```diff
@@ -480,32 +483,36 @@ class QueryService {
                 highDimensionResourceService.createAssayConstraint([ids: assayIds], AssayConstraint.ASSAY_ID_LIST_CONSTRAINT)
         ]
 
-        Map<HighDimensionDataTypeResource, Collection<Assay>> assaysByType =
-                highDimensionResourceService.getSubResourcesAssayMultiMap(oldAssayConstraints)
-
-        if (assaysByType.size() == 0) {
-            throw new InvalidQueryException("Unknown high dimensional data type.")
-        }
-        else if (assaysByType.size() > 1) {
-            throw new InvalidQueryException("Expected only one high dimensional data type. Got ${assaysByType.keySet()*.dataTypeName}")
-        }
-
-        def assayByType = assaysByType.iterator().next()
-
-        def platformList = assayByType.value*.platform as Set
-        if (platformList.size() != 1){
-            throw new InvalidQueryException("Result assays contain different platforms: ${platformList*.id}")
+        HighDimensionDataTypeResource typeResource
+        if(type == 'autodetect') {
+            Map<HighDimensionDataTypeResource, Collection<Assay>> assaysByType =
+                    highDimensionResourceService.getSubResourcesAssayMultiMap(oldAssayConstraints)
+            if (assaysByType.size() == 1) {
```

Comment:
The assaysByType map can not be empty, that could only happen if there is no high dim data loaded at all, and in that case the previous step at line 480 already returns an empty result.